_RegexParser fails to build with a 5.6 toolchain

When doing a `--bootstrapping=hosttools` Swift compiler build with Xcode 13.4, you get the error:

```
/Users/hamish/src/swift-dev/swift-experimental-string-processing/Sources/_RegexParser/Utility/TypeConstruction.swift:63:64: error: cannot infer return type for closure with multiple statements; add explicit type to disambiguate
    let result = elementTypes.withContiguousStorageIfAvailable { elementTypesBuffer in
                                                               ^
                                                                                    -> <#Result#>
```

We should ideally allow _RegexParser to compile with a 5.6 toolchain.

--- a/Sources/_RegexParser/Utility/TypeConstruction.swift
+++ b/Sources/_RegexParser/Utility/TypeConstruction.swift
@@ -60,7 +60,7 @@ public enum TypeConstruction {
       flags |= 0x10000
     }
     
-    let result = elementTypes.withContiguousStorageIfAvailable { elementTypesBuffer in
+    let result = elementTypes.withContiguousStorageIfAvailable { elementTypesBuffer -> (value: Any.Type, state: Int) in
       if let labels = labels {
         return labels.withCString { labelsPtr in
           swift_getTupleTypeMetadata(

